libafl_qemu/src/injections.rs

```diff
@@ -0,0 +1,312 @@
+                        let func_pc = find_function(emu, &lib.name, &func.function, lib.off)
+                            .unwrap_or_default();
+                        if func_pc > 0 {
+                            println!("Function {} found at {func_pc:#x}", func.function);
+                            let data: u64 = (id << 8) + u64::from(func.parameter);
+                            let _hook_id = emu.set_hook(data, func_pc, on_call_check, false);
+                            found = 1;
+                        }
+                    }
+                }
+                if found > 0 {
+                    for test in &target.tests {
+                        tokens.push(test.input_value.clone());
+                    }
+                } else {
+                    println!("Function not found: {}", func.function);
+                }
+            }
+            id += 1;
+        }
+
+        TOKENS.set(tokens.clone()).expect("Failed to set tokens");
+        tokens
+    }
+}
+
+impl<S> QemuHelper<S> for QemuInjectionHelper
+where
+    S: UsesInput,
+{
+    fn init_hooks<QT>(&self, hooks: &QemuHooks<QT, S>)
+    where
+        QT: QemuHelperTuple<S>,
+    {
+        hooks.syscalls(Hook::Function(syscall_hook::<QT, S>));
+    }
+}
+
+fn syscall_hook<QT, S>(
+    hooks: &mut QemuHooks<QT, S>, // our instantiated QemuHooks
+    _state: Option<&mut S>,
+    syscall: i32, // syscall number
+    x0: u64,      // registers ...
+    x1: u64,
+    _x2: u64,
+    _x3: u64,
+    _x4: u64,
+    _x5: u64,
+    _x6: u64,
+    _x7: u64,
+) -> SyscallHookResult
+where
+    QT: QemuHelperTuple<S>,
+    S: UsesInput,
+{
+    //println!("syscall_hook {} {}", syscall, SYS_execve);
+    debug_assert!(i32::try_from(SYS_execve).is_ok());
+    if syscall == SYS_execve as i32 {
+        let _helper = hooks
+            .helpers_mut()
+            .match_first_type_mut::<QemuInjectionHelper>()
+            .unwrap();
+        if x0 > 0 && x1 > 0 {
+            let c_array = x1 as *const *const c_char;
+            let cmd = unsafe {
+                let c_str_ptr = x0 as *const c_char;
+                CStr::from_ptr(c_str_ptr).to_string_lossy()
+            };
+            assert!(
+                cmd.to_lowercase() != "fuzz",
+                "Found verified command injection!"
+            );
+            //println!("CMD {}", cmd);
+
+            let first_parameter = unsafe {
+                if (*c_array.offset(1)).is_null() {
+                    return SyscallHookResult::new(None);
+                }
+                CStr::from_ptr(*c_array.offset(1)).to_string_lossy()
+            };
+            let second_parameter = unsafe {
+                if (*c_array.offset(2)).is_null() {
+                    return SyscallHookResult::new(None);
+                }
+                CStr::from_ptr(*c_array.offset(2)).to_string_lossy()
+            };
+            if first_parameter == "-c"
+                && (second_parameter.to_lowercase().contains("';fuzz;'")
```

Comment:
Why is this hardcoded? I though it's all configurable?

Reply:
for command injections it makes no sense to be configurable as we hook a syscall, not tons of libc etc functions.

Reply:
I don't think this is false-positive-free btw:
`-c` `echo "\";fuzz;"` would trigger(falsely)